File: radar/src/lib/color.ts
import { Color, type PlayerData, type Team } from "./data";

export const COLORS = {
    backdrop: "#18181c",
    base: "#1e1e23",
    highlight: "#32323c",
    subtext: "#8c8c8c",
    text: "#ffffff",
    red: "#f06464",
    orange: "#f08c5a",
    yellow: "#f0c878",
    green: "#a0f082",
    teal: "#50c8c8",
    blue: "#6496f0",
    purple: "#b478f0",
} as const;

export function playerColor(player: PlayerData): string {
    switch (player.color) {
        case Color.Yellow:
            return COLORS.yellow;
        case Color.Purple:
            return COLORS.purple;
        case Color.Green:
            return COLORS.green;
        case Color.Blue:
            return COLORS.blue;
        case Color.Orange:
            return COLORS.orange;
        default:
            return COLORS.subtext;
    }
}

export function teamColor(team: Team): string {
    switch (team) {
        case "T":
            return COLORS.orange;
        case "CT":
            return COLORS.blue;
        default:
            return COLORS.subtext;
    }
}

export function healthColor(health: number, maxHealth: number): string {
    const maximum = Math.max(1, maxHealth);
    const percent = Math.max(0, Math.min(1, health / maximum));

    if (percent <= 0.5) {
        const yellowPercent = percent * 200;
        return `color-mix(in srgb, ${COLORS.red} ${100 - yellowPercent}%, ${COLORS.yellow})`;
    }

    const greenPercent = (percent - 0.5) * 200;
    return `color-mix(in srgb, ${COLORS.yellow} ${100 - greenPercent}%, ${COLORS.green})`;
}

File: cheat/src/ui/color.rs
#![allow(unused)]
use egui::Color32;
use serde::{Deserialize, Serialize};

pub struct Colors;

impl Colors {
    pub const BACKDROP: Color32 = Color32::from_rgb(15, 15, 20);
    pub const BASE: Color32 = Color32::from_rgb(22, 22, 30);
    pub const HIGHLIGHT: Color32 = Color32::from_rgb(40, 40, 55);
    pub const SUBTEXT: Color32 = Color32::from_rgb(160, 160, 175);
    pub const TEXT: Color32 = Color32::from_rgb(245, 245, 250);
    pub const RED: Color32 = Color32::from_rgb(255, 75, 110);
    pub const ORANGE: Color32 = Color32::from_rgb(255, 140, 60);
    pub const YELLOW: Color32 = Color32::from_rgb(255, 210, 80);
    pub const GREEN: Color32 = Color32::from_rgb(50, 240, 140);
    pub const TEAL: Color32 = Color32::from_rgb(0, 220, 220);
    pub const BLUE: Color32 = Color32::from_rgb(80, 160, 255);
    pub const PURPLE: Color32 = Color32::from_rgb(190, 90, 255);

    pub const ACCENT_COLORS: [(&str, Color32); 7] = [
        ("Red", Self::RED),
        ("Orange", Self::ORANGE),
        ("Yellow", Self::YELLOW),
        ("Green", Self::GREEN),
        ("Teal", Self::TEAL),
        ("Blue", Self::BLUE),
        ("Purple", Self::PURPLE),
    ];
}
